moveit_experimental/jog_arm/src/jog_arm/low_pass_filter.cpp

```diff
@@ -50,28 +50,22 @@ void LowPassFilter::reset(double data)
 {
   previous_measurements_[0] = data;
   previous_measurements_[1] = data;
-  previous_measurements_[2] = data;
 
-  previous_filtered_measurements_[0] = data;
-  previous_filtered_measurements_[1] = data;
+  previous_filtered_measurement_ = data;
 }
 
 double LowPassFilter::filter(double new_measurement)
 {
   // Push in the new measurement
-  previous_measurements_[2] = previous_measurements_[1];
   previous_measurements_[1] = previous_measurements_[0];
   previous_measurements_[0] = new_measurement;
 
   double new_filtered_msrmt =
-      (1. / (1. + filter_coeff_ * filter_coeff_ + 1.414 * filter_coeff_)) *
-      (previous_measurements_[2] + 2. * previous_measurements_[1] + previous_measurements_[0] -
-       (filter_coeff_ * filter_coeff_ - 1.414 * filter_coeff_ + 1.) * previous_filtered_measurements_[1] -
-       (-2. * filter_coeff_ * filter_coeff_ + 2.) * previous_filtered_measurements_[0]);
+      (1. / (1. + filter_coeff_)) * (previous_measurements_[1] + previous_measurements_[0] -
```

Comment:
Should be (Matlab syntax):

new_filtered_msrmt = (1 + c / (2*pi))^-1 * (previous_measurements_[1] + previous_measurements_[0] - (1 - c/(2*pi)) * previous_filtered_measurement_);